df.map.xml

```diff
@@ -1079,25 +1109,25 @@
         </enum-item>
     </enum-type>
 
-    <bitfield-type type-name='construction_flags' base-type='uint8_t'>
-        <flag-bit name='no_build_item' comment='build item is created from scratch upon removing construction, does not exist beforehand'/>
-        <flag-bit name='top_of_wall' comment='used on the floors above constructed walls so you cannot remove them'/>
+    <bitfield-type type-name='construction_flags' base-type='uint8_t'> bay12: EVENT_CONSTRUCTION_FLAG_*
+        <flag-bit name='no_build_item' comment='bay12: MUST_GENERATE_ITEMS; build item is created from scratch upon removing construction, does not exist beforehand'/>
+        <flag-bit name='top_of_wall' comment='bay12: ITEMLESS_CEILING; used on the floors above constructed walls so you cannot remove them'/>
     </bitfield-type>
 
     <struct-type type-name='construction' instance-vector='$global.world.constructions' key-field='pos' original-name='event_constructionst' custom-methods='true'>
         <compound name='pos' type-name='coord'/>
-        <enum type-name='item_type' base-type='int16_t' name='item_type'/>
+        <enum type-name='item_type' name='item_type'/>
         <int16_t name='item_subtype' refers-to='(item-subtype-target $$._parent.item_type $)'/>
         <int16_t name='mat_type' ref-target='material' aux-value='$$.mat_index'/>
         <int32_t name='mat_index'/>
         <bitfield type-name='construction_flags' name='flags'/>
-        <enum base-type='int16_t' type-name='tiletype' name='original_tile'/>
+        <enum base-type='uint16_t' type-name='tiletype' name='original_tile'/>
     </struct-type>
 
-    <enum-type type-name='flow_type' base-type='int16_t'>
+    <enum-type type-name='flow_type'> bay12: FlowTypes, no base type
```

Comment:
removing the `base-type='int16_t'` from this type causes `flow_info` to generate the following header:
```cpp
  struct DFHACK_EXPORT flow_info {
    enum_field<df::flow_type,int16_t> type;
```
instead of
```cpp
  struct DFHACK_EXPORT flow_info {
    df::flow_type type;
```

Reply:
Let me merge this and fix in the patches I have prepared. We can discuss this more on Discord